draft-ietf-tls-esni.md

```diff
@@ -216,64 +239,83 @@ ECHConfigs structure.
+`ECHConfig` structure with a key using a KEM they do not support.
 
 cipher_suites
-: The list of HPKE {{!I-D.irtf-cfrg-hpke}} AEAD and KDF identifier pairs clients
-can use for encrypting the ClientHello.
+: The list of HPKE AEAD and KDF identifier pairs clients can use for encrypting
+ClientHelloInner.
 
 maximum_name_length
-: The largest name the server expects to support, if known. If this value is
-not known it can be set to zero, in which case clients SHOULD use the inner
-ClientHello padding scheme described below. That could happen if wildcard names
-are in use, or if names can be added or removed from the anonymity set during
-the lifetime of a particular resource record value.
+: The largest name the server expects to support, if known. If this value is not
+known it can be set to zero, in which case clients SHOULD use the padding scheme
+described in {{padding}}. This is useful when wildcard names are in use, or if
+names can be added or removed from the anonymity set during the lifetime of a
+particular resource record value.
 
 extensions
-: A list of extensions that the client can take into consideration when
-generating a ClientHello message. The purpose of the field is to provide room
-for additional functionality in the future. See {{config-extensions}} for
-guidance on what type of extensions are appropriate for this structure.
-
-The format is defined in {{RFC8446}}, Section 4.2. The same interpretation rules
-apply: extensions MAY appear in any order, but there MUST NOT be more than one
-extension of the same type in the extensions block. An extension can be tagged
-as mandatory by using an extension type codepoint with the high order bit set to
-1. A client which receives a mandatory extension they do not understand MUST
-reject the ECHConfig content.
+: A list of extensions that the client must take into consideration when
+generating a ClientHello message. These are described below
+({{config-extensions}}).
+
+## Configuration Extensions {#config-extensions}
+
+ECH configuration extensions are used to to provide room for additional
+functionality as needed. See {{config-extensions-guidance}} for guidance on
+which types of extensions are appropriate for this structure.
+
+The format is as defined in {{RFC8446}}, Section 4.2.
+The same interpretation rules apply: extensions MAY appear in any order, but
+there MUST NOT be more than one extension of the same type in the extensions
+block. An extension can be tagged as mandatory by using an extension type
+codepoint with the high order bit set to 1. A client that receives a mandatory
+extension they do not understand MUST reject the `ECHConfig` content.
 
 Clients MUST parse the extension list and check for unsupported mandatory
 extensions. If an unsupported mandatory extension is present, clients MUST
-reject the ECHConfig value.
+ignore the `ECHConfig`.
+
+## Versions {#versioning}
+
+[[TODO: Replace this with an enumerated type? This is what's required for
```

Comment:
I don't think this is necessary. The version field speaks for itself, and this section doesn't add much. Can we remove this?

Reply:
Yes, I'll revert this. However, it seems technically incorrect to me: using RFC 8446 as the authoritative standard, the `select` block is only defined for enumerated types. This might be splitting hairs though ... I'm not sure how strict RFCs are about this.

Reply:
Reverted.